### DISCUSSION TITLE
Low probabilities for .approximate_distribution with embedding model

### DISCUSSION THREAD

**Discussion by BH-Ma** [MaartenGr/BERTopic#1294] [Q&A] [answered]
Hi Maarten,

Thanks a lot for the awesome package, it does help me a lot with my research project with **online reviews for some apps.** I would much appreciate it if you could suggest me the following points:

1. Currently, I encounter an issue when I calculate the topic distributions with `.approximate_distribution(use_embedding_model=True)`, the resulting probabilities across all topics are very low (<0.1). 
I am wondering if I shall adjust the `min_similarity` parameter to a lower value (such as _0.01_).

2. I actually want to convert the **topic probabilities** to **binary variables that describe** **the existence of topics in the document**. I am doing this by setting a threshold value for probabilities. Do you have any suggestions on how to set the threshold? Currently, I tried different values, and intuitively set it to _0.2_.

3. It seems that the `approximate_distribution` commend can provide some **incorrect predictions**, especially on some small-sized topics. Although I understand this is something inevitable, do you have any suggestions on how to improve it?

Thanks for your time! ;)

**Comment by MaartenGr** [comment, marked as answer]
> Currently, I encounter an issue when I calculate the topic distributions with .approximate_distribution(use_embedding_model=True), the resulting probabilities across all topics are very low (<0.1). I am wondering if I shall adjust the min_similarity parameter to a lower value (such as 0.01).

The probability values that you get depend on a number of things, including the `stride`, `window`, and `use_embedding_model` parameters. The reason for this is that if you have a small stride and window, then many small aggregations are created which are likely to have smaller probabilities since they contain less information than their bigger alternatives. For instance, increasing the window often helps in creating more accurate representations and if you increase the stride similarly, then that reduces computational costs. 

In other words, setting the `min_similarity` lower is a fine solution but you can also look into the parameters I mentioned.  

> I actually want to convert the topic probabilities to binary variables that describe the existence of topics in the document. I am doing this by setting a threshold value for probabilities. Do you have any suggestions on how to set the threshold? Currently, I tried different values, and intuitively set it to 0.2.

That depends highly on the distribution for your probabilities and there is no fixed value for that. You would have to evaluate yourself what a good threshold would be. 

> It seems that the approximate_distribution commend can provide some incorrect predictions, especially on some small-sized topics. Although I understand this is something inevitable, do you have any suggestions on how to improve it?

It depends on what you mean with **incorrect predictions**. Incorrect can mean that they do not match with the underlying clustering algorithm but that does not necessarily mean that they are incorrect or do not properly represent the topic they match. One thing you can do is using `use_embedding_model` to use the embedding model which may be closer to the original underlying clustering algorithm.

**Reply by BH-Ma** [reply]
Thanks a lot for your very prompt help, will try it out with your suggestions!
